File: app/src/main/java/me/ycdev/android/devtools/root/RootJarExecutor.java
package me.ycdev.android.devtools.root;

import java.util.Arrays;

import me.ycdev.android.devtools.utils.AppConfigs;
import me.ycdev.android.lib.common.internalapi.android.app.ActivityManagerIA;
import timber.log.Timber;

public class RootJarExecutor {
    private static final String TAG = "RootJarExecutor";
    private static final boolean DEBUG = AppConfigs.DEBUG_LOG;

    public static final String CMD_FORCE_STOP_PACKAGE = "amForceStop";

    private String[] mArgs;

    public static void main(String[] args) {
        if (DEBUG) Timber.tag(TAG).d("Received params: " + Arrays.toString(args));
        if (args.length < 1) {
            if (DEBUG) Timber.tag(TAG).e("Usage: RootJarExecutor <command> [command parameters]");
            return;
        }

        new RootJarExecutor(args).execute();
    }

    private RootJarExecutor(String[] args) {
        mArgs = args;
    }

    private void execute() {
        String cmd = mArgs[0];
        if (cmd.equals(CMD_FORCE_STOP_PACKAGE)) {
            if (mArgs.length > 1) {
                String filePath = mArgs[1];
                forceStopPackage(filePath);
            } else {
                Timber.tag(TAG).e("Usage: RootJarExecutor amForceStop <file path>");
            }
        }
    }

    public static void forceStopPackage(String params) {
        String[] pkgNames = params.split("#");
        Object service = ActivityManagerIA.INSTANCE.getIActivityManager();
        if (service != null) {
            if (DEBUG) Timber.tag(TAG).d("to kill apps: %s", Arrays.toString(pkgNames));
            for (String pkg : pkgNames) {
                ActivityManagerIA.INSTANCE.forceStopPackage(service, pkg);
            }
        }
    }
}

File: app/src/main/java/me/ycdev/android/devtools/root/RootJarExecutor.kt
package me.ycdev.android.devtools.root

import me.ycdev.android.lib.common.internalapi.android.app.ActivityManagerIA
import timber.log.Timber

class RootJarExecutor private constructor(private val args: Array<String>) {
    private fun execute() {
        val cmd = args[0]
        if (cmd == CMD_FORCE_STOP_PACKAGE) {
            if (args.size > 1) {
                val filePath = args[1]
                forceStopPackage(filePath)
            } else {
                Timber.tag(TAG).e("Usage: RootJarExecutor amForceStop <file path>")
            }
        }
    }

    companion object {
        private const val TAG = "RootJarExecutor"

        const val CMD_FORCE_STOP_PACKAGE = "amForceStop"

        @JvmStatic
        fun main(args: Array<String>) {
            Timber.tag(TAG).d(
                "Received params: %s", args.contentToString()
            )
            if (args.isEmpty()) {
                Timber.tag(TAG).e("Usage: RootJarExecutor <command> [command parameters]")
                return
            }
            RootJarExecutor(args).execute()
        }

        fun forceStopPackage(params: String) {
            val pkgNames = params.split("#").toTypedArray()
            val service = ActivityManagerIA.getIActivityManager()
            if (service != null) {
                Timber.tag(TAG).d(
                    "to kill apps: %s",
                    pkgNames.contentToString()
                )
                for (pkg in pkgNames) {
                    ActivityManagerIA.forceStopPackage(service, pkg)
                }
            }
        }
    }
}